libs/variant/include/variant/variant.hpp

```diff
@@ -576,4 +584,150 @@ void Variant::IterateObject(Function const &function) const
+    serializer << typecode;
+    switch (var.type_)
+    {
+    case Type::Type::UNDEFINED:
+    {
+      break;
+    }
+
+    case Type::Type::NULL_VALUE:
+    {
+      break;
+    }
+    case Type::Type::INTEGER:
+    {
+      serializer << var.primitive_.integer;
+      break;
+    }
+    case Type::Type::FLOATING_POINT:
+    {
+      serializer << var.primitive_.float_point;
+      break;
+    }
+    case Type::Type::FIXED_POINT:
+    {
+      serializer << var.primitive_.integer;
+      break;
+    }
+    case Type::Type::BOOLEAN:
+    {
+      serializer << var.primitive_.boolean;
+      break;
+    }
+    case Type::Type::STRING:
+    {
+      serializer << var.string_;
+      break;
+    }
+    default:
+    {
+      throw std::runtime_error{"Variant has unknown type."};
+    }
+    case Type::Type::ARRAY:
+    {
+      auto sz = static_cast<uint32_t>(var.array_.size());
+      serializer << sz;
+      for (auto const &element : var.array_)
+      {
+        Serialize(serializer, element);
+      }
+      break;
+    }
+    case Type::Type::OBJECT:
+    {
+      auto sz = static_cast<uint32_t>(var.object_.size());
+      serializer << sz;
+      for (auto const &element : var.object_)
+      {
+        serializer << element.first;
+        Serialize(serializer, *element.second);
+      }
+      break;
+    }
+    }
+  }
+
+  template <typename Deserializer>
+  static void Deserialize(Deserializer &deserializer, Type &var)
+  {
+    int typecode;
+    deserializer >> typecode;
+    auto type = static_cast<Type::Type>(typecode);
+    var.type_ = type;
+    switch (var.type_)
+    {
+    case Type::Type::UNDEFINED:
+      break;
+    case Type::Type::NULL_VALUE:
+      break;
+    case Type::Type::INTEGER:
+      deserializer >> var.primitive_.integer;
+      break;
+    case Type::Type::FLOATING_POINT:
+      deserializer >> var.primitive_.float_point;
+      break;
+    case Type::Type::FIXED_POINT:
+      deserializer >> var.primitive_.integer;
+      break;
+    case Type::Type::BOOLEAN:
+      deserializer >> var.primitive_.boolean;
+      break;
+    case Type::Type::STRING:
+      deserializer >> var.string_;
+      break;
+    case Type::Type::ARRAY:
+    {
+      uint32_t    count_tmp;
+      std::size_t count;
+      deserializer >> count_tmp;
+      count = count_tmp;
+      for (std::size_t i = 0; i < count; i++)
+      {
+        Type v;
+        Deserialize(deserializer, v);
+        var.array_.push_back(v);
+      }
+      break;
+    }
+    case Type::Type::OBJECT:
+    {
+      uint32_t    count_tmp;
+      std::size_t count;
+      deserializer >> count_tmp;
+      count = count_tmp;
+      for (std::size_t i = 0; i < count; i++)
+      {
+        Type                       v;
+        byte_array::ConstByteArray k;
+        deserializer >> k;
+        Deserialize(deserializer, v);
+        var[k] = v;
+      }
+      break;
+    }
+    default:
+    {
+      throw std::runtime_error{"Variant has unknown type."};
+    }
```

Comment:
should be relocated after the switch(...){...} block and  change `break;` to `return;`

Reply:
done